File: booking-service/src/main/java/com/hotelbooking/entity/Room.java
package com.hotelbooking.entity;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Entity
@Table(name = "rooms", schema = "BOOKING_HOTELS_SCHEMA")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@EqualsAndHashCode
@ToString
public class Room {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private int number;
    @ManyToOne
    private Hotel hotel;
    @ManyToOne
    private RoomCategory roomCategory;
    private int price;
    private int persons;
}

File: client-angular/src/app/model/entity/room.ts
import {Country} from './country';
import {City} from './city';
import {RoomCategory} from './roomCategory';
import {Hotel} from './hotel';

export class Room {
  id: number;
  number: number;
  hotel: Hotel;
  roomCategory: RoomCategory;
  price: number;
  persons: number;
}
